### PR TITLE
Add section on JavaScript interop for custom types

### PR DESCRIPTION
**Pull request by nino**
For #200.

Does something like this look reasonable?

### CODE REVIEW

File: book-src/tour/custom-types.md [hunk 1]
```diff
@@ -250,3 +250,23 @@ LoggedIn("Kim")
...
+Guest
+LoggedIn("Kim")
+Cat(name: "Jinx", cuteness: 2002)
+```
+
+```javascript
+// JavaScript
+new Guest();              // => {}
+new LoggedIn("Kim");      // => { 0: "Kim" }
+new Animal("Jinx", 2002); // => { name: "Jinx", cuteness: 2002 }
```
Comment (lpil) [inline book-src/tour/custom-types.md:271]: Could you remove these comments? They're not accurate, and I think the constructors on the left are clear enough.

**Review by lpil — COMMENTED** [review]
Nice! Do you think we should also show them being imported?

File: book-src/tour/custom-types.md (hunk 1 above)
Comment (nino) [reply book-src/tour/custom-types.md:271]: Yeah, I was trying to find a way to explain how to access named and unnamed properties. Maybe we could do something like

```javascript
const user = new LoggedIn("Kim");
user[0] === "Kim";

const cat = new Cat("Jinx", 2002);
cat.cuteness === 2002;
```

Not sure if that's overkill

**Comment by nino** [comment]
> Do you think we should also show them being imported?

That's a good idea actually

**Review by lpil — APPROVED** [review]
Wonderful! Thank you

**Pull request merged**